packages/core/src/delegation/DelegationNode.ts

```diff
@@ -193,8 +158,21 @@ export default class DelegationNode extends DelegationBaseNode
    * @returns Promise containing a SubmittableExtrinsic.
    */
   public async revoke(identity: Identity): Promise<SubmittableExtrinsic> {
-    log.debug(`:: revoke(${this.id})`)
-    return revoke(this.id, identity)
+    const { steps, node } = await this.isDelegating(identity.address)
+    if (!node) {
+      throw SDKErrors.ERROR_UNAUTHORIZED(
+        `Identity with address ${identity.address} is not among the delegators and may not revoke this node`
+      )
+    }
```

Comment:
this check is optional, but given that we are required to traverse the whole tree which means retrieving the information whether we can revoke or not, it seems dumb to spam the network with a transaction that is doomed to fail.